translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java

```diff
@@ -51,18 +54,26 @@ public void translate(List<TranslationObj> translationObjs) throws TranslationEx
     
     List<TranslationObj> toTranslate = translationObjs.stream().filter(
         t -> t.getTranslation() == null).toList();
-    
-    if(toTranslate.isEmpty()) {
-      //all entries retrieved from cache, processing complete
-      return;
+
+    if(!toTranslate.isEmpty()) {    
+      translationService.translate(toTranslate);
+      if(isCachingEnabled()) {
+        //save result in the redis cache
+        redisCacheService.store(toTranslate);  
+      }
     }
     
-    translationService.translate(toTranslate);
-    
-    if(isCachingEnabled()) {
-      //save result in the redis cache
-      redisCacheService.store(toTranslate);  
+    //logging the number of translated/cached lines and chars
+    int numLinesCached=(int) translationObjs.stream().filter(el -> el.isRetrievedFromCache()).count();
+    int numCharsCached=translationObjs.stream().filter(el -> el.isRetrievedFromCache()).map(el -> el.getTranslation().length()).reduce(0, Integer::sum);
+    int numLinesTranslated=toTranslate.size();
+    int numCharsTranslated=toTranslate.stream().map(el -> el.getText().length()).reduce(0, Integer::sum);
```

Comment:
this is incorectly computed, as it must not include the texts discarded for translation (i.e. nonTranslatable & same language)